### PR TITLE
Inventory protocol improvements

### PR DESCRIPTION
**Pull request by Jim3Things**
Update the observer to use json parsing and interface definitions derived from the proto files.
Some improvements to the cluster and blade details displays to carry more of the detail information.
Incorporate fix from jj-0416.

### CODE REVIEW

File: pkg/protos/inventory/capacity.ts_ref [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+ *
+ * The multi-dimensionality is important for schedulers to understand, as the
+ * exhaustion of one dimension can lead to a case where the other unused
+ * capacity dimensions are stranded.  For instance, if all cores are consumed
+ * then no free memory, disk, or other dimensions of a blade can be used.
+ */
+
+/** Define the set of known accelerators, such as GPUs or FPGAs. */
+export interface Accelerator {
+  v100: Accelerator_NVIDIAV100 | undefined;
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/capacity.ts_ref:18]: It means that the element may be missing -- it is like the '?' on a C# declaration.

File: pkg/protos/inventory/external.proto [hunk 2]
```diff
@@ -30,15 +31,17 @@ message External {
...
     message Rack {
+        RackDetails details = 1;
+
         // This assumes a single overhead item per rack.  May want to allow multiple to handle
         // subdivisions for power or network, say.
 
-        Pdu pdu = 1;
-        Tor tor = 2;
+        Pdu pdu = 2;
+        Tor tor = 3;
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/external.proto:40]: I've intended to pick it up in a later change.

File: pkg/protos/inventory/external.ts_ref [hunk 3]
```diff
@@ -0,0 +1,357 @@
+/* eslint-disable */
+import {
+  RackDetails,
+  ZoneDetails,
+} from "../../../../../../github.com/Jim3Things/CloudChamber/pkg/protos/inventory/common";
+import { BladeCapacity } from "../../../../../../github.com/Jim3Things/CloudChamber/pkg/protos/inventory/capacity";
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply pkg/protos/inventory/external.ts_ref:6]: This is part of the generated stuff.  if you look at external.tsx, you'll see a more sane path...

File: pkg/protos/inventory/external.proto (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply pkg/protos/inventory/external.proto:40]: Makes sense.

File: pkg/protos/inventory/capacity.ts_ref (hunk 1 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply pkg/protos/inventory/capacity.ts_ref:18]: Ok.

File: pkg/protos/inventory/external.ts_ref (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply pkg/protos/inventory/external.ts_ref:6]: Ok.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**